addons/account/wizard/accrued_orders.py

```diff
@@ -135,11 +142,8 @@ def _ellipsis(string, size):
         for order in orders:
             if len(orders) == 1 and self.amount:
                 total_balance = self.amount
-                order_line = order.order_line[0]
-                if is_purchase:
-                    account = order_line.product_id.property_account_expense_id or order_line.product_id.categ_id.property_account_expense_categ_id
-                else:
-                    account = order_line.product_id.property_account_income_id or order_line.product_id.categ_id.property_account_income_categ_id
+                order_line = order.order_line.filtered(lambda l: not l.display_type)[0]
```

Comment:
But why the `[0]`? And why a special case when there is only one order?

Reply:
the [0] is for the reason I explained above. Now, for the special case of len(orders) == 1:

Let's assume an order with 0 quantity received, if we select only this order, we are able to select an amount in the wizard and then we pass in the condition to select the account linked to the first order line.
If we select orders in a batch, we're not allow to chose any amount, therefore they will be no line on the wizard for the order with 0 quantity received.
If we delete this check and do only the instructions in else statement, there will be no line on the wizard when setting the amount manually.